x/gov/handler.go

```diff
@@ -248,14 +251,20 @@ func updateElectorateExecutor(elecBucket *ElectorateBucket) TallyResultExecutor
 			return errors.Wrap(errors.ErrHuman, "details not set")
 		}
 		// merge diff
-		elect, err := elecBucket.GetElectorate(db, proposal.ElectorateID)
+		_, obj, err := elecBucket.GetLatestVersion(db, proposal.ElectorateRef.ID)
 		if err != nil {
 			return errors.Wrap(err, "failed to load electorate")
 		}
+		elect, err := asElectorate(obj)
+		if err != nil {
+			return err
+		}
 		merger := newMerger(elect.Electors)
-		merger.merge(details.DiffElectors)
+		if err := merger.merge(details.DiffElectors); err != nil {
+			return errors.Wrap(err, "merge failed")
+		}
 		elect.Electors, elect.TotalElectorateWeight = merger.serialize()
-		if err := elecBucket.Save(db, orm.NewSimpleObj(proposal.ElectorateID, elect)); err != nil {
+		if _, err := elecBucket.Update(db, proposal.ElectorateRef.ID, elect); err != nil {
```

Comment:
Okay, I see why you have the three arguments now.

You use the same ID in `GetLatestVersion` as in `Update`.
And in the meantime just modify `elect`, which is the de-serialized value.

If I understand correctly, `obj.Key() == proposal.ElectorateRef.ID` but it is cleaner how you do it here.

Reply:
`obj.Key() == VersionedIDRef.Marshalled` containing the unique ID (from the sequence) and the Version (which is stored with the object and incremented in the bucket)
`GetLatestVersion` returns the current VersionedIDRef as well but it is ignored in this function as all information are available already.

Reply:
Thanks for the clarity.
The flow works nice as it is.